python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java

```diff
@@ -230,34 +230,32 @@ public static int firstParameterOffset(FunctionSymbol functionSymbol, boolean is
     return 0;
   }
 
-  public static Optional<FunctionSymbol> getOverriddenMethod(FunctionSymbol functionSymbol) {
+  public static List<FunctionSymbol> getOverriddenMethods(FunctionSymbol functionSymbol) {
     Symbol owner = ((FunctionSymbolImpl) functionSymbol).owner();
     if (owner == null || owner.kind() != CLASS) {
-      return Optional.empty();
+      return List.of();
     }
     ClassSymbol classSymbol = (ClassSymbol) owner;
-    if (classSymbol.superClasses().isEmpty()) {
-      return Optional.empty();
-    }
-    for (Symbol superClass : classSymbol.superClasses()) {
-      if (superClass.kind() == CLASS) {
-        Optional<FunctionSymbol> overriddenSymbol = ((ClassSymbol) superClass).resolveMember(functionSymbol.name())
-          .flatMap(SymbolUtils::getFunctionSymbol);
-        if (overriddenSymbol.isPresent()) {
-          return overriddenSymbol;
-        }
-      }
-    }
-    return Optional.empty();
+
+    return classSymbol.superClasses()
+      .stream()
+      .filter(ClassSymbol.class::isInstance)
+      .map(ClassSymbol.class::cast)
+      .map(c -> c.resolveMember(functionSymbol.name())
+        .map(SymbolUtils::getFunctionSymbols)
+        .orElseGet(List::of))
+      .filter(Predicate.not(List::isEmpty))
+      .findFirst()
+      .orElseGet(List::of);
   }
 
-  public static Optional<FunctionSymbol> getFunctionSymbol(@Nullable Symbol symbol) {
+  public static List<FunctionSymbol> getFunctionSymbols(@Nullable Symbol symbol) {
```

Comment:
I kind of feel like we should keep a singular version of this method as well as `getOverridenMethod` because from an API perspective it feels a bit surprising to expect a list of function symbols from a symbol, knowing you'd get a list only if there is an ambiguous symbol with only functions that match whatever disambiguation strategy is selected.